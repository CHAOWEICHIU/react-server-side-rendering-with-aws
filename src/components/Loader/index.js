import React from 'react'
import styled, { keyframes } from 'styled-components'

function animation (props) {
  return keyframes`
    0% {
     transform: rotate(0deg);
    }
    100% {
       transform: rotate(360deg);
    }
  `
}

const Loader = styled.div`
  align-self: center;
  margin-top: 100px;
  border: 12px solid #055877;
  border-right-color: transparent;
  border-radius: 50%;
  display: inline-block;
  height: 100px;
  width: 100px;
  animation: ${props => animation} 1s ease-in-out infinite;
`

export default Loader
