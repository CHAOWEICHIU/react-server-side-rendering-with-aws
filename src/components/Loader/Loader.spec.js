import React            from 'react'
import { shallow }      from 'enzyme'
import renderer         from 'react-test-renderer'
import Loader           from './'
import 'jest-styled-components'

it('<Loader />', ()=>{
  const tree = renderer.create(
    <Loader />
  ).toJSON()
  expect(tree).toMatchStyledComponentsSnapshot()
})
