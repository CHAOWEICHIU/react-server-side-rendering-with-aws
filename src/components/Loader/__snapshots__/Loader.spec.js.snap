// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> 1`] = `
.cMStMu {
  align-self: center;
  -ms-flex-item-align: center;
  -webkit-align-self: center;
  margin-top: 100px;
  border: 12px solid #055877;
  border-right-color: transparent;
  border-radius: 50%;
  display: inline-block;
  height: 100px;
  width: 100px;
  animation: iECmZH 1s ease-in-out infinite;
  -webkit-animation: iECmZH 1s ease-in-out infinite;
}

<div
  className="cMStMu"
/>
`;
